toolchain/parse/node_kind.def

```diff
@@ -215,9 +215,24 @@ CARBON_PARSE_NODE_KIND_BRACKET(LibraryDecl, LibraryIntroducer,
 // LibrarySpecifier
 CARBON_PARSE_NODE_KIND_CHILD_COUNT(LibrarySpecifier, 1, Library)
 
+// Declaration names.
+//
+//   _repeated_: NameQualifier
+//   _external_: IdentifierName
+//   _optional_ _external_: ImplicitParamList
+//   _optional_ _external_: TuplePattern
+// _declaration name_
```

Comment:
This comment is unclear to me: it offers a tree structure that, as far as I can tell, does not exist. Had you considered a non-tree-like representation?

Reply:
How would you like this to be presented?

Alternatively: how would you feel about removing all of these structure comments now that we have `typed_nodes.h` that describes the structure?

Reply:
Removing all these comments seems fair.